asyncstatsd/metrics/basic.py

```diff
@@ -0,0 +1,61 @@
+from dataclasses import dataclass, field
+from datetime import timedelta
+
+
+@dataclass
+class StatsdMetric:
+    name: str
+    value: int | float | timedelta
+    unit: str
+    rate: float = field(default=1.0)
+
+    def get_value(self) -> str:
+        return f"{self.value}"
+
+    def serialize(self) -> str | None:
+        value = self.get_value()
+        serialized = f"{self.name}:{value}|{self.unit}"
```

Comment:
it seems `:` and `|` have special meaning in the format used for serialisation. Does `name`/`unit` need to and/or get validated to not contain those characters?

Reply:
yeah, good point! At this stage they are not validated but probably should..